docs-web/configuration/ingress-resources/advanced-configuration-with-snippets.md

```diff
@@ -0,0 +1,107 @@
+
+## Using Snippets
+
+The example below shows how to use snippets to customize the NGINX configuration template using annotations.
+```yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: cafe-ingress-with-snippets
+  annotations:
+    nginx.org/server-snippets: |
+      location / {
+          return 302 /coffee;
+      }
+    nginx.org/location-snippets: |
+      add_header my-test-header test-value;
+spec:
+  rules:
+  - host: cafe.example.com
+    http:
+      paths:
+      - path: /tea
+        backend:
+          serviceName: tea-svc
+          servicePort: 80
+      - path: /coffee
+        backend:
+          serviceName: coffee-svc
+          servicePort: 80
+```
+
+Generated NGINX configuration:
+```nginx
+server {
+    listen 80;
+
+
+    location / {
+        return 302 /coffee;
+    }
+
+
+    location /coffee {
+        proxy_http_version 1.1;
+
+
+        add_header my-test-header test-value;
+        ...
+        proxy_pass http://default-cafe-ingress-with-snippets-cafe.example.com-coffee-svc-80;
+    }
+
+    location /tea {
+        proxy_http_version 1.1;
+        
+        add_header my-test-header test-value;
+        ...
+        proxy_pass http://default-cafe-ingress-with-snippets-cafe.example.com-tea-svc-80;
+    }
+}
+```
+**Note**: The generated configs are truncated for the clarity of the example.
+
+## Summary of Snippets
+
+See the [snippets annotations](/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-annotations/#snippets-and-custom-templates) documentation for more information.
+
+## Disadvantages of using snippets
+
+* *Complexity*. To use snippets, you will need to:
+  * Understand NGINX configuration primitives and implement a correct NGINX configuration.
+  * Understand how the IC generates NGINX configuration so that a snippet doesn't interfere with the other features in the configuration.
+* *Decreased robustness*. An incorrect snippet makes the NGINX config invalid, which causes reload failures. This will prevent any new configuration updates, including updates for the other Ingress resources, until the snippet is fixed.
+* *Security implications*. Snippets give access to NGINX configuration primitives and those primitives are not validated by the Ingress Controller. For example, a snippet can configure NGINX to serve the TLS certificates and keys used for TLS termination for Ingress resources.
+
+> **Note**: If the NGINX config includes an invalid snippet, NGINX will continue to operate with the latest valid configuration.
+
+## Troubleshooting
+
+* If a snippet includes an invalid NGINX configuration, the Ingress Controller will fail to reload NGINX. The error will be reported in the Ingress Controller logs and an event with the error will be associated with the Ingress resource:
```

Comment:
this list only have one element. I think it's redundant because of that. Could we just remove the list and make its content regular paragraphs?